console.go

```diff
@@ -64,87 +71,140 @@ func (c *consoleHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	l.mu.Lock()
 	logCount := l.logCount
 	maxSeverity := l.maxSeverity
+	attributes := l.reqAttributes
 	l.mu.Unlock()
 
 	// status code should also set the minimum maxSeverity to Error
 	if sw.Status() > 399 && maxSeverity < logging.Error {
 		maxSeverity = logging.Error
 	}
 
-	l.consolef(
-		maxSeverity, severityColor(maxSeverity), "%s %s %d %s requestSize=%d responseSize=%d logCount=%d",
-		r.Method, r.URL.Path, sw.Status(), time.Since(begin), requestSize(r.Header.Get("Content-Length")), sw.length, logCount,
+	msg := fmt.Sprintf("%s %s %d %s %s=%d %s=%d %s=%d", r.Method, r.URL.Path, sw.Status(), time.Since(begin),
+		cslReqSize, requestSize(r.Header.Get("Content-Length")), cslRespSize, sw.length, cslLogCount, logCount,
 	)
+	for k, v := range attributes {
+		msg += fmt.Sprintf(" %s=%v", k, v)
+	}
+	l.console(maxSeverity, severityColor(maxSeverity), msg)
 }
 
 type consoleLogger struct {
-	r           *http.Request
-	noColor     bool
-	mu          sync.Mutex
-	maxSeverity logging.Severity
-	logCount    int
+	root          *consoleLogger
+	r             *http.Request
+	noColor       bool
+	rsvdReqKeys   []string
+	attributes    map[string]any // attributes for child (trace) logs
+	mu            sync.Mutex
+	maxSeverity   logging.Severity
+	logCount      int
+	reqAttributes map[string]any // attributes for the parent request log
 }
 
 // newConsoleLogger logs all output to console
 func newConsoleLogger(r *http.Request, noColor bool) *consoleLogger {
-	return &consoleLogger{r: r, noColor: noColor, maxSeverity: logging.Debug}
+	l := &consoleLogger{
+		r: r, noColor: noColor,
+		rsvdReqKeys:   []string{cslReqSize, cslRespSize, cslLogCount},
+		maxSeverity:   logging.Debug,
+		reqAttributes: make(map[string]any),
+		attributes:    make(map[string]any),
+	}
+	l.root = l // root is self
+
+	return l
+}
+
+// newChild returns a new child consoleLogger
+func (l *consoleLogger) newChild() *consoleLogger {
+	return &consoleLogger{
+		root:          l.root,
+		r:             l.r,
+		noColor:       l.noColor,
+		rsvdReqKeys:   l.rsvdReqKeys,
+		maxSeverity:   logging.Debug,
+		attributes:    make(map[string]any),
+		reqAttributes: nil, // stating zero value for clear intent
```

Comment:
I think we should state what this clear intent is so there are no mistries for the reader


```suggestion
		reqAttributes: nil, // reqAttributes is only used in the root logger, never the child.
```

Reply:
That's a great point. Done